crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs

```diff
@@ -220,16 +305,108 @@ impl BuiltInResult {
         }
     }
 
-    pub fn to_c(&self) -> &'static str {
+    pub fn to_c(&self) -> String {
+        if self.is_custom_result_type() {
+            return format!(
+                "struct {}$Result{}And{}",
+                SWIFT_BRIDGE_PREFIX, self.ok_ty_string, self.err_ty_string
+            );
+        }
         // TODO: Choose the kind of Result representation based on whether or not the ok and error
         //  types are primitives.
         //  See `swift-bridge/src/std_bridge/result`
         if self.ok_ty.can_be_encoded_with_zero_bytes() {
-            "struct __private__ResultVoidAndPtr"
+            format!("struct __private__ResultVoidAndPtr")
         } else {
-            "struct __private__ResultPtrAndPtr"
+            format!("struct __private__ResultPtrAndPtr")
         }
     }
+
+    pub fn generate_ffi_definition(
+        &self,
+        swift_bridge_path: &Path,
+        types: &TypeDeclarations,
+    ) -> Option<TokenStream> {
+        if self.is_custom_result_type() {
```

Comment:
We can make the early return path more clear, as well as remove the indentation here by doing:

```rust
if !self.is_custom_result_type {
    return None;
}

// ...
```

Reply:
How about making the early return in the others:

For example: 

https://github.com/NiwakaDev/swift-bridge/blob/8214e585cd222c2bfa03047997c191fcf55233e2/crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs#L205-L241

Reply:
The one that you linked to has logic in both the true and false case, so it seems fine as is.